proto/streamlit/proto/Navigation.proto

```diff
@@ -24,8 +24,17 @@ import "streamlit/proto/AppPage.proto";
 message Navigation {
   repeated string sections = 1;
   repeated AppPage app_pages = 2;
-  string position = 3;
+  Position position = 3;
 
   // The script hash for the page identified by st.navigation
   string page_script_hash = 4;
+
+  // Position of the Navigation
+  enum Position {
+    // Plain, fixed width text.
+    SIDEBAR = 0;
+
+    // Shows an error message.
+    HIDDEN = 1;
```

Comment:
extreme nit: I might flip these and make `hidden` the 0, just in case there are other positions being added in the future and then its hidden=0, sidebar=1, bottom=2 etc.